3fgripper:
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50  

# ratating platform ---------------------------------------
  l_finger_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: l_finger_joint
    pid: {p: 350, i: 0.1, d: 200}
  r_finger_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: r_finger_joint
    pid: {p: 350, i: 0.1, d: 200}

# finger left ---------------------------------------
  l_joint1_position_controller:
    type: effort_controllers/JointPositionController
    joint: l_Joint1
    pid: {p: 5000.0, i: 0.01, d: 500.0}
  l_joint2_position_controller:
    type: effort_controllers/JointPositionController
    joint: l_Joint2
    pid: {p: 5000.0, i: 0.01, d: 500.0}
  l_joint3_position_controller:
    type: effort_controllers/JointPositionController
    joint: l_Joint3
    pid: {p: 5000.0, i: 0.01, d: 500.0}

# finger middle ---------------------------------------
  m_joint1_position_controller:
    type: effort_controllers/JointPositionController
    joint: m_Joint1
    pid: {p: 5000.0, i: 0.01, d: 500.0}
  m_joint2_position_controller:
    type: effort_controllers/JointPositionController
    joint: m_Joint2
    pid: {p: 5000.0, i: 0.01, d: 500.0}
  m_joint3_position_controller:
    type: effort_controllers/JointPositionController
    joint: m_Joint3
    pid: {p: 5000.0, i: 0.01, d: 500.0}

# finger right ---------------------------------------
  r_joint1_position_controller:
    type: effort_controllers/JointPositionController
    joint: r_Joint1
    pid: {p: 5000.0, i: 0.01, d: 500.0}
  r_joint2_position_controller:
    type: effort_controllers/JointPositionController
    joint: r_Joint2
    pid: {p: 5000.0, i: 0.01, d: 500.0}
  r_joint3_position_controller:
    type: effort_controllers/JointPositionController
    joint: r_Joint3
    pid: {p: 5000.0, i: 0.01, d: 500.0}

  
